internal/commands/untrust_builder_test.go

```diff
@@ -0,0 +1,148 @@
+	var (
+		logger       logging.Logger
+		outBuf       bytes.Buffer
+		tempPackHome string
+		configPath   string
+	)
+
+	it.Before(func() {
+		var err error
+
+		logger = ilogging.NewLogWithWriters(&outBuf, &outBuf)
+
+		tempPackHome, err = ioutil.TempDir("", "pack-home")
+		h.AssertNil(t, err)
+		h.AssertNil(t, os.Setenv("PACK_HOME", tempPackHome))
+
+		configPath = filepath.Join(tempPackHome, "config.toml")
+	})
+
+	it.After(func() {
+		h.AssertNil(t, os.Unsetenv("PACK_HOME"))
+		h.AssertNil(t, os.RemoveAll(tempPackHome))
+	})
+
+	when("#UntrustBuilder", func() {
+		when("no builder is provided", func() {
+			it("prints usage", func() {
+				command := untrustBuilderCommandWithTrustedBuilders(t, logger, configPath)
+				command.SetArgs([]string{})
+				h.AssertNil(t, command.Execute())
+				h.AssertContains(t, outBuf.String(), "Usage:")
+			})
+		})
+
+		when("builder is already trusted", func() {
+			it("removes builder from the config", func() {
+				builderName := "some-builder"
+
+				command := untrustBuilderCommandWithTrustedBuilders(t, logger, configPath, builderName)
+				command.SetArgs([]string{builderName})
+
+				h.AssertNil(t, command.Execute())
+
+				b, err := ioutil.ReadFile(configPath)
+				h.AssertNil(t, err)
+				h.AssertNotContains(t, string(b), builderName)
+
+				h.AssertContains(t,
+					outBuf.String(),
+					fmt.Sprintf("Builder %s is no longer trusted", style.Symbol(builderName)),
+				)
+			})
+
+			it("removes only the named builder when multiple builders are trusted", func() {
+				untrustBuilder := "stop/trusting:me"
+				stillTrustedBuilder := "very/safe/builder"
+
+				command := untrustBuilderCommandWithTrustedBuilders(t,
+					logger,
+					configPath,
+					untrustBuilder,
+					stillTrustedBuilder,
+				)
+				command.SetArgs([]string{untrustBuilder})
+
+				h.AssertNil(t, command.Execute())
+
+				b, err := ioutil.ReadFile(configPath)
+				h.AssertNil(t, err)
+				h.AssertContains(t, string(b), stillTrustedBuilder)
+				h.AssertNotContains(t, string(b), untrustBuilder)
+			})
+		})
+
+		when("builder wasn't already trusted", func() {
+			it("does nothing and reports builder wasn't trusted", func() {
+				neverTrustedBuilder := "never/trusted-builder"
+				stillTrustedBuilder := "very/safe/builder"
+
+				command := untrustBuilderCommandWithTrustedBuilders(t, logger, configPath, stillTrustedBuilder)
+				command.SetArgs([]string{neverTrustedBuilder})
+
+				h.AssertNil(t, command.Execute())
+
+				b, err := ioutil.ReadFile(configPath)
+				h.AssertNil(t, err)
+				h.AssertContains(t, string(b), stillTrustedBuilder)
+				h.AssertNotContains(t, string(b), neverTrustedBuilder)
+
+				h.AssertContains(t,
+					outBuf.String(),
+					fmt.Sprintf("Builder %s wasn't trusted", style.Symbol(neverTrustedBuilder)),
+				)
+			})
+		})
+	})
+}
+
+//nolint:whitespace // A leading line of whitespace is left after a method declaration with multi-line arguments
+func untrustBuilderCommandWithTrustedBuilders(
```

Comment:
This function seems unnecessary if we were to instead just have a function "newConfigWithTrustedBuilders":

Then usages would be two lines but would be more understandable IMO:

```go
cfg := newConfigWithTrustedBuilders(builders)
command := commands.UntrustBuilder(logger, cfg)
```

Reply:
Agreed, updated. Used a small manager struct to avoid repeatedly passing around `t` & `configPath` that aren't really relevant to the test cases.